packages/core/src/fhirlexer/tokenize.ts

```diff
@@ -48,7 +48,7 @@ export class Tokenizer {
     this.keywords = keywords;
     this.operators = operators;
     this.dateTimeLiterals = !!options?.dateTimeLiterals;
-    this.symbolRegex = options?.symbolRegex ?? /[$\w]/;
+    this.symbolRegex = options?.symbolRegex ?? /[$\w%]/;
```

Comment:
1. Should variables use `SymbolAtom` or should we create a new `VariableAtom`?  No strong opinion on that
2. We should probably update the regex so that the `%` can only be at the beginning

We should make sure that we can handle expressions like `a%b`.  I believe (?) that should translate to `a % b` where `%` is a modulo operator.  But it might parse as a single symbol with value `a%b`?

Reply:
I'm in favor of a single `SymbolAtom` that covers this case.  The FHIRPath grammar actually calls these "external constants", so calling our atom "variable" might be slightly misaligned with respect to that

Reply:
Actually.... looking at the way that we use `this.symbolRegex`, there's no good way to specify start of string.  That pushes me toward thinking we should add `VariableAtom` and a dedicated parsing handler.

Reply:
Actually @jamestouri  - Now that I'm looking at the spec, our `$` is also incorrect.

Maybe we can do the simple thing for now as you've done it, and then we can do a fast follow to implement `$` and `%` prefixes properly in a separate commit.

Here is the official grammar: https://github.com/HL7/FHIRPath/blob/master/spec/fhirpath.g4

My reading is that the `$` not only must be at the start of the string, it also is only valid for a select number of keywords:

```
invocation                          // Terms that can be used after the function/member invocation '.'
        : identifier                                            #memberInvocation
        | function                                              #functionInvocation
        | '$this'                                               #thisInvocation
        | '$index'                                              #indexInvocation
        | '$total'                                              #totalInvocation
        ;
```